// @classdo/classdo/dist/clients/OrganizationMembers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var Viewer_1 = require("../models/Viewer");
/** Client to call OrganizatoinMembers schema */
var OrganizationMembersClient = /** @class */ (function () {
    /** @ignore */
    function OrganizationMembersClient(client) {
        this.client = client;
    }
    /**
     * Get list of OrganizationMember.
     *
     * ```typescript
     * client.organizationMembers.list(['id']).then(v => {
     *   console.log(v)
     * })
     * ```
     *
     * @param fields Array of [[OrganizationMember]] key names. Returns specified fields as result.
     * @param args
     * @param option
     */
    OrganizationMembersClient.prototype.list = function (fields, args, option) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var result;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.client.query({
                            viewer: Viewer_1.buildViewerQuery(['id'], {
                                members: { fields: fields, args: args, with: option }
                            })
                        })];
                    case 1:
                        result = _a.sent();
                        return [2 /*return*/, {
                                errors: result.errors,
                                data: result.errors ? null : result.data.viewer.members
                            }];
                }
            });
        });
    };
    /**
     * Get a OrganizationMember.
     *
     * ```typescript
     * client.organizationMembers.get(['id'], 'ck4l8tsef03nn0726825k3k9t').then(v => {
     *   console.log(v.data?.id)
     * })
     * ```
     *
     * @param fields Array of [[OrganizationMember]] key names. Returns specified fields as result.
     * @param id Organization member id.
     * @param option
     */
    OrganizationMembersClient.prototype.get = function (fields, id, option) {
        var _a, _b;
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var result, data;
            return tslib_1.__generator(this, function (_c) {
                switch (_c.label) {
                    case 0: return [4 /*yield*/, this.client.query({
                            viewer: Viewer_1.buildViewerQuery(['id'], {
                                members: { fields: fields, args: { input: { where: { id: id } } }, with: option }
                            })
                        })];
                    case 1:
                        result = _c.sent();
                        data = ((_b = (_a = result.data.viewer.members) === null || _a === void 0 ? void 0 : _a.edges[0]) === null || _b === void 0 ? void 0 : _b.node) || null;
                        if (!data) {
                            throw new Error("not found organization member: " + id);
                        }
                        return [2 /*return*/, { errors: result.errors, data: data }];
                }
            });
        });
    };
    return OrganizationMembersClient;
}());
exports.OrganizationMembersClient = OrganizationMembersClient;
