// @classdo/classdo/dist/clients/Roles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var Viewer_1 = require("../models/Viewer");
/** Client to call OrganizationMemberRoles schema */
var RolesClient = /** @class */ (function () {
    function RolesClient(client) {
        this.client = client;
    }
    /**
     * Get list of OrganizationMemberRoles.
     *
     * ```
     * const rolesResult = client.roles.list(['id', 'name']).then(v => {
     *   console.log(v)
     * })
     * ```
     *
     * @param fields Array of [[OrganizationMemberRole]] key names. Returns specified fields as result.
     * @param args
     */
    RolesClient.prototype.list = function (fields, args) {
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var result;
            return tslib_1.__generator(this, function (_a) {
                switch (_a.label) {
                    case 0: return [4 /*yield*/, this.client.query({
                            viewer: Viewer_1.buildViewerQuery(['id'], { roles: { fields: fields, args: args } })
                        })];
                    case 1:
                        result = _a.sent();
                        return [2 /*return*/, {
                                errors: result.errors,
                                data: result.errors ? null : result.data.viewer.roles
                            }];
                }
            });
        });
    };
    /**
     * Get a role.
     *
     * ```
     * client.roles.get(['id'], 'xxxxxxxxxxx').then(v => {
     *   console.log(v)
     * })
     * ```
     *
     * @param fields Array of [[OrganizationMemberRole]] key names. Returns specified fields as result.
     * @param id OrganizationMemberRole id.
     */
    RolesClient.prototype.get = function (fields, id) {
        var _a, _b;
        return tslib_1.__awaiter(this, void 0, void 0, function () {
            var result, data;
            return tslib_1.__generator(this, function (_c) {
                switch (_c.label) {
                    case 0: return [4 /*yield*/, this.client.query({
                            viewer: Viewer_1.buildViewerQuery(['id'], { roles: { fields: fields } })
                        })];
                    case 1:
                        result = _c.sent();
                        data = ((_b = (_a = result.data.viewer.roles) === null || _a === void 0 ? void 0 : _a.edges[0]) === null || _b === void 0 ? void 0 : _b.node) || null;
                        if (!data) {
                            throw new Error("not found role: " + id);
                        }
                        return [2 /*return*/, { errors: result.errors, data: data }];
                }
            });
        });
    };
    return RolesClient;
}());
exports.RolesClient = RolesClient;
