// @classdo/classdo/dist/generated/graphql.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var ContactType;
(function (ContactType) {
    ContactType["PhoneNumber"] = "PhoneNumber";
    ContactType["Email"] = "Email";
})(ContactType = exports.ContactType || (exports.ContactType = {}));
var EmailOrderByInput;
(function (EmailOrderByInput) {
    EmailOrderByInput["IdAsc"] = "id_ASC";
    EmailOrderByInput["IdDesc"] = "id_DESC";
    EmailOrderByInput["CreatedAtAsc"] = "createdAt_ASC";
    EmailOrderByInput["CreatedAtDesc"] = "createdAt_DESC";
    EmailOrderByInput["UpdatedAtAsc"] = "updatedAt_ASC";
    EmailOrderByInput["UpdatedAtDesc"] = "updatedAt_DESC";
    EmailOrderByInput["ValueAsc"] = "value_ASC";
    EmailOrderByInput["ValueDesc"] = "value_DESC";
})(EmailOrderByInput = exports.EmailOrderByInput || (exports.EmailOrderByInput = {}));
var InvitationStatus;
(function (InvitationStatus) {
    InvitationStatus["Pending"] = "Pending";
    InvitationStatus["Accepted"] = "Accepted";
    InvitationStatus["Cancelled"] = "Cancelled";
})(InvitationStatus = exports.InvitationStatus || (exports.InvitationStatus = {}));
var LedgerType;
(function (LedgerType) {
    LedgerType["Use"] = "Use";
    LedgerType["UseRecording"] = "UseRecording";
    LedgerType["Topup"] = "Topup";
    LedgerType["AutoTopup"] = "AutoTopup";
    LedgerType["AddFreeCredit"] = "AddFreeCredit";
})(LedgerType = exports.LedgerType || (exports.LedgerType = {}));
var Locale;
(function (Locale) {
    Locale["En"] = "En";
    Locale["Ja"] = "Ja";
})(Locale = exports.Locale || (exports.Locale = {}));
var OrganizationMemberOrderByInput;
(function (OrganizationMemberOrderByInput) {
    OrganizationMemberOrderByInput["IdAsc"] = "id_ASC";
    OrganizationMemberOrderByInput["IdDesc"] = "id_DESC";
    OrganizationMemberOrderByInput["CreatedAtAsc"] = "createdAt_ASC";
    OrganizationMemberOrderByInput["CreatedAtDesc"] = "createdAt_DESC";
    OrganizationMemberOrderByInput["UpdatedAtAsc"] = "updatedAt_ASC";
    OrganizationMemberOrderByInput["UpdatedAtDesc"] = "updatedAt_DESC";
})(OrganizationMemberOrderByInput = exports.OrganizationMemberOrderByInput || (exports.OrganizationMemberOrderByInput = {}));
var OrganizationMemberRoleOrderByInput;
(function (OrganizationMemberRoleOrderByInput) {
    OrganizationMemberRoleOrderByInput["IdAsc"] = "id_ASC";
    OrganizationMemberRoleOrderByInput["IdDesc"] = "id_DESC";
    OrganizationMemberRoleOrderByInput["NameAsc"] = "name_ASC";
    OrganizationMemberRoleOrderByInput["NameDesc"] = "name_DESC";
})(OrganizationMemberRoleOrderByInput = exports.OrganizationMemberRoleOrderByInput || (exports.OrganizationMemberRoleOrderByInput = {}));
var PermissionAction;
(function (PermissionAction) {
    PermissionAction["OrganizationEdit"] = "Organization_Edit";
    PermissionAction["OrganizationDelete"] = "Organization_Delete";
    PermissionAction["OrganizationMembersShow"] = "OrganizationMembers_Show";
    PermissionAction["OrganizationMemberRolesEdit"] = "OrganizationMemberRoles_Edit";
    PermissionAction["BillingEdit"] = "Billing_Edit";
    PermissionAction["RoomsCreate"] = "Rooms_Create";
    PermissionAction["RoomsEdit"] = "Rooms_Edit";
    PermissionAction["RoomsDelete"] = "Rooms_Delete";
    PermissionAction["VideochatActivate"] = "Videochat_Activate";
    PermissionAction["VideochatToggleRemoteAudio"] = "Videochat_ToggleRemoteAudio";
    PermissionAction["VideochatToggleRemoteVideo"] = "Videochat_ToggleRemoteVideo";
    PermissionAction["RoomStatusBePresenter"] = "RoomStatus_BePresenter";
    PermissionAction["RoomRecordingsCreate"] = "RoomRecordings_Create";
    PermissionAction["RoomRecordingsShow"] = "RoomRecordings_Show";
    PermissionAction["RoomMembersDelete"] = "RoomMembers_Delete";
    PermissionAction["RoomMembersInviteContacts"] = "RoomMembers_InviteContacts";
    PermissionAction["RoomMembersInviteOrganizationMembers"] = "RoomMembers_InviteOrganizationMembers";
    PermissionAction["RoomMembersInviteQuickJoin"] = "RoomMembers_InviteQuickJoin";
    PermissionAction["ApiKeyManagement"] = "APIKey_Management";
})(PermissionAction = exports.PermissionAction || (exports.PermissionAction = {}));
var RoomMemberOrderByInput;
(function (RoomMemberOrderByInput) {
    RoomMemberOrderByInput["IdAsc"] = "id_ASC";
    RoomMemberOrderByInput["IdDesc"] = "id_DESC";
    RoomMemberOrderByInput["CreatedAtAsc"] = "createdAt_ASC";
    RoomMemberOrderByInput["CreatedAtDesc"] = "createdAt_DESC";
    RoomMemberOrderByInput["UpdatedAtAsc"] = "updatedAt_ASC";
    RoomMemberOrderByInput["UpdatedAtDesc"] = "updatedAt_DESC";
})(RoomMemberOrderByInput = exports.RoomMemberOrderByInput || (exports.RoomMemberOrderByInput = {}));
var RoomOrderByInput;
(function (RoomOrderByInput) {
    RoomOrderByInput["IdAsc"] = "id_ASC";
    RoomOrderByInput["IdDesc"] = "id_DESC";
    RoomOrderByInput["CreatedAtAsc"] = "createdAt_ASC";
    RoomOrderByInput["CreatedAtDesc"] = "createdAt_DESC";
    RoomOrderByInput["UpdatedAtAsc"] = "updatedAt_ASC";
    RoomOrderByInput["UpdatedAtDesc"] = "updatedAt_DESC";
    RoomOrderByInput["NameAsc"] = "name_ASC";
    RoomOrderByInput["NameDesc"] = "name_DESC";
    RoomOrderByInput["DescriptionAsc"] = "description_ASC";
    RoomOrderByInput["DescriptionDesc"] = "description_DESC";
    RoomOrderByInput["ArchivedAtAsc"] = "archivedAt_ASC";
    RoomOrderByInput["ArchivedAtDesc"] = "archivedAt_DESC";
    RoomOrderByInput["OldIdAsc"] = "oldId_ASC";
    RoomOrderByInput["OldIdDesc"] = "oldId_DESC";
    RoomOrderByInput["ExitRoomLinkAsc"] = "exitRoomLink_ASC";
    RoomOrderByInput["ExitRoomLinkDesc"] = "exitRoomLink_DESC";
})(RoomOrderByInput = exports.RoomOrderByInput || (exports.RoomOrderByInput = {}));
