// @classdo/classdo/dist/models/Billing.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var typed_graphqlify_1 = require("typed-graphqlify");
var TestUtils_1 = require("../TestUtils");
var Billing_1 = require("./Billing");
describe('buildBillingQuery', function () {
    it('should build a query', function () {
        var result = Billing_1.buildBillingQuery(['id', 'year', 'month'], { input: { year: 2020, month: 3 } }, {
            records: {
                fields: ['id', 'date'],
                with: {
                    usages: {
                        fields: ['paidSec', 'amount'],
                        with: {
                            user: {
                                fields: ['id'], with: { profile: { fields: ['firstName'] } }
                            },
                            room: { fields: ['name'] }
                        }
                    },
                    freeCredits: { fields: ['paidSec', 'amount'] },
                    topups: { fields: ['paidSec', 'amount'] },
                }
            }
        });
        TestUtils_1.compareGraphqlQuery(typed_graphqlify_1.query(result), "\n      query(input: { year: 2020, month: 3 }) {\n        id\n        year\n        month\n        records {\n          id\n          date\n          usages {\n            paidSec\n            amount\n            user {\n              id\n              profile {\n                firstName\n              }\n            }\n            room {\n              name\n            }\n          }\n          freeCredits {\n            paidSec\n            amount\n          }\n          topups {\n            paidSec\n            amount\n          }\n        }\n      }\n    ");
    });
});
