// @classdo/classdo/dist/models/Emails.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var typed_graphqlify_1 = require("typed-graphqlify");
var Utils_1 = require("../Utils");
exports.Email = {
    id: typed_graphqlify_1.types.string,
    value: typed_graphqlify_1.types.string
};
/**
 * @ignore
 */
var buildEmailEdge = function (email) { return ({
    node: email,
    cursor: typed_graphqlify_1.types.string
}); };
/**
 * @ignore
 */
var buildEmails = function (args, email) {
    var emails = {
        totalCount: typed_graphqlify_1.types.number,
        pageInfo: {
            hasNextPage: typed_graphqlify_1.types.boolean,
            hasPreviousPage: typed_graphqlify_1.types.boolean,
            startCursor: typed_graphqlify_1.types.optional.string,
            endCursor: typed_graphqlify_1.types.optional.string
        },
        edges: [buildEmailEdge(email)]
    };
    return args ? typed_graphqlify_1.params(Utils_1.preprocessArgs(args), emails) : emails;
};
/**
 * @ignore
 */
function buildEmailsQuery(fields, args) {
    var pickedField = Utils_1.pick(exports.Email, fields);
    return buildEmails(args, pickedField);
}
exports.buildEmailsQuery = buildEmailsQuery;
