// @classdo/classdo/dist/models/Invitations.test.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var Invitations_1 = require("./Invitations");
var graphql_1 = require("../generated/graphql");
var typed_graphqlify_1 = require("typed-graphqlify");
var TestUtils_1 = require("../TestUtils");
describe('buildInvitationQuery', function () {
    it('should build a query', function () {
        var result = Invitations_1.buildInvitationQuery(['id', 'contactFullName']);
        TestUtils_1.compareGraphqlQuery(typed_graphqlify_1.query(result), "\n      query {\n        id\n        contactFullName\n      }\n    ");
    });
});
describe('#buildSendInvitationMutation', function () {
    it('should build a query', function () {
        var result = Invitations_1.buildSendInvitationMutation(['id', 'contactFullName'], {
            data: {
                contactFullName: 'invitee',
                contactType: graphql_1.ContactType.Email,
                contactInfo: 'example@test.com',
                locale: graphql_1.Locale.En,
                organizationMemberRoleId: 'role1',
                roomId: 'room1'
            }
        });
        TestUtils_1.compareGraphqlQuery(typed_graphqlify_1.mutation(result), "\n        mutation {\n          sendInvitation(data: {\n            contactFullName: \"invitee\",\n            contactType: Email,\n            contactInfo: \"example@test.com\",\n            locale: En,\n            organizationMemberRoleId: \"role1\",\n            roomId: \"room1\"\n          }) {\n            id\n            contactFullName\n          }\n        }\n      ");
    });
});
