// @classdo/classdo/dist/models/OrganizationMembers.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var typed_graphqlify_1 = require("typed-graphqlify");
var Utils_1 = require("../Utils");
var Users_1 = require("./Users");
var OrganizationMemberRoles_1 = require("./OrganizationMemberRoles");
exports.OrganizationMember = {
    id: typed_graphqlify_1.types.string
};
/**
 * @ignore
 */
var buildOrganizationMemberEdge = function (organizationMember) { return ({
    node: organizationMember,
    cursor: typed_graphqlify_1.types.string
}); };
/**
 * @ignore
 */
var buildOrganizationMembers = function (args, organizationMember) {
    var roomMembers = {
        totalCount: typed_graphqlify_1.types.number,
        pageInfo: {
            hasNextPage: typed_graphqlify_1.types.boolean,
            hasPreviousPage: typed_graphqlify_1.types.boolean,
            startCursor: typed_graphqlify_1.types.optional.string,
            endCursor: typed_graphqlify_1.types.optional.string
        },
        edges: [buildOrganizationMemberEdge(organizationMember)]
    };
    return args ? typed_graphqlify_1.params(Utils_1.preprocessArgs(args), roomMembers) : roomMembers;
};
/**
 * @ignore
 */
function buildOrganizationMembersQuery(fields, args, option) {
    var pickedField = Utils_1.pick(exports.OrganizationMember, fields);
    if (option && option.user) {
        pickedField['user'] = Users_1.buildUserQuery(option.user.fields, option.user.with || {});
    }
    if (option && option.role) {
        pickedField['role'] = OrganizationMemberRoles_1.buildOrganizationMemberRoleQuery(option.role.fields);
    }
    return buildOrganizationMembers(args, pickedField);
}
exports.buildOrganizationMembersQuery = buildOrganizationMembersQuery;
