// @classdo/classdo/dist/models/OrganizationMemberRoles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var typed_graphqlify_1 = require("typed-graphqlify");
var Utils_1 = require("../Utils");
exports.OrganizationMemberRole = {
    id: typed_graphqlify_1.types.string,
    name: typed_graphqlify_1.types.string
};
/**
 * @ignore
 */
var buildOrganizationMemberRoleEdge = function (role) { return ({
    node: role,
    cursor: typed_graphqlify_1.types.string
}); };
/**
 * @ignore
 */
var buildOrganizationMemberRoles = function (args, role) {
    var roomMembers = {
        totalCount: typed_graphqlify_1.types.number,
        pageInfo: {
            hasNextPage: typed_graphqlify_1.types.boolean,
            hasPreviousPage: typed_graphqlify_1.types.boolean,
            startCursor: typed_graphqlify_1.types.optional.string,
            endCursor: typed_graphqlify_1.types.optional.string
        },
        edges: [buildOrganizationMemberRoleEdge(role)]
    };
    return args ? typed_graphqlify_1.params(Utils_1.preprocessArgs(args), roomMembers) : roomMembers;
};
/**
 * @ignore
 */
function buildOrganizationMemberRoleQuery(fields) {
    var pickedField = Utils_1.pick(exports.OrganizationMemberRole, fields);
    return pickedField;
}
exports.buildOrganizationMemberRoleQuery = buildOrganizationMemberRoleQuery;
/**
 * @ignore
 */
function buildOrganizationMemberRolesQuery(fields, args) {
    var pickedField = Utils_1.pick(exports.OrganizationMemberRole, fields);
    return buildOrganizationMemberRoles(args, pickedField);
}
exports.buildOrganizationMemberRolesQuery = buildOrganizationMemberRolesQuery;
