// @classdo/classdo/dist/models/UserProfiles.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var tslib_1 = require("tslib");
var typed_graphqlify_1 = require("typed-graphqlify");
var Utils_1 = require("../Utils");
var Emails_1 = require("./Emails");
exports.UserProfile = {
    id: typed_graphqlify_1.types.string,
    firstName: typed_graphqlify_1.types.string,
    lastName: typed_graphqlify_1.types.string,
    phoneNumber: typed_graphqlify_1.types.string
};
/**
 * @ignore
 */
function resolveOption(option) {
    var emails = option.emails
        ? { emails: Emails_1.buildEmailsQuery(option.emails.fields, option.emails.args) }
        : {};
    return tslib_1.__assign({}, emails);
}
/**
 * @ignore
 */
function buildUserProfileQuery(fields, option) {
    var pickedField = Utils_1.pick(exports.UserProfile, fields);
    var resolvedOption = option ? resolveOption(option) : {};
    return tslib_1.__assign(tslib_1.__assign({}, pickedField), resolvedOption);
}
exports.buildUserProfileQuery = buildUserProfileQuery;
