// @classdo/classdo/dist/models/Viewer.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var typed_graphqlify_1 = require("typed-graphqlify");
var Rooms_1 = require("./Rooms");
var Utils_1 = require("../Utils");
var OrganizationMembers_1 = require("./OrganizationMembers");
var OrganizationMemberRoles_1 = require("./OrganizationMemberRoles");
var Billing_1 = require("./Billing");
var Viewer = {
    id: typed_graphqlify_1.types.string,
    name: typed_graphqlify_1.types.string,
    logoImageUrl: typed_graphqlify_1.types.string
};
/**
 * @ignore
 */
function buildViewerQuery(fields, option) {
    var pickedFields = Utils_1.pick(Viewer, fields || []);
    if (option && option.rooms) {
        pickedFields['rooms'] = Rooms_1.buildRoomsQuery(option.rooms.fields, option.rooms.args, option.rooms.with || {});
    }
    if (option && option.members) {
        pickedFields['members'] = OrganizationMembers_1.buildOrganizationMembersQuery(option.members.fields, option.members.args, option.members.with || {});
    }
    if (option && option.roles) {
        pickedFields['roles'] = OrganizationMemberRoles_1.buildOrganizationMemberRolesQuery(option.roles.fields, option.roles.args);
    }
    if (option && option.billing) {
        pickedFields['billing'] = Billing_1.buildBillingQuery(option.billing.fields, option.billing.args, option.billing.with);
    }
    return pickedFields;
}
exports.buildViewerQuery = buildViewerQuery;
